### ISSUE TITLE
SOGo cannot log in due to the "/" in the user name of the mailbox

### ISSUE THREAD

**Issue by rawvoid** [mailcow/mailcow-dockerized#5190]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/mailcow/mailcow-dockerized/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... I understand that not following the below instructions will result in immediate closure and/or deletion of my issue.
- [X] ... I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [X] ... I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] ... I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

### Description

```plain text
hello everyone,

I added a mailbox with a username containing a "/" symbol, but it cannot log in in SOGo, and the browser shows a 403 error code. 

Does mailcow have user name validation? Usernames containing "/" symbols should not be allowed.

Thanks.
```

### Logs:

```plain text
-
```

### Steps to reproduce:

```plain text
-
```

### Which branch are you using?

master

### Operating System:

Ubuntu 22.04 LTS

### Server/VM specifications:

4 cores, 4G memory

### Is Apparmor, SELinux or similar active?

no

### Virtualization technology:

KVM

### Docker version:

-

### docker-compose version or docker compose version:

-

### mailcow version:

2023-04a

### Reverse proxy:

Nginx

### Logs of git diff:

```plain text
-
```

### Logs of iptables -L -vn:

```plain text
-
```

### Logs of ip6tables -L -vn:

```plain text
-
```

### Logs of iptables -L -vn -t nat:

```plain text
-
```

### Logs of ip6tables -L -vn -t nat:

```plain text
-
```

### DNS check:

```plain text
-
```

**Comment by FreddleSpl0it** [comment]
mailcow uses the PHP filter `FILTER_VALIDATE_EMAIL` to validate email addresses, which follows the RFC 822. This specification allows certain special characters, including `/`  in the local-part of an email address. I think using "/" in an email address is just causing troubles currently. Maybe we should discuss if we want to accept the kinds of characters or not.

@mkuron @dragoangel what do you think?

**Comment by dragoangel** [comment]
I agree that such input should be treated as invalid

**Comment by mkuron** [comment]
Yeah, `/` in an email address is definitely not a good idea, even if RFC822 allows it.

**Comment by rawvoid** [comment]
The cause of the problem may be that the '/' character in the URL redirected by SOGo after successful login is not URL-encoded. Maybe this problem needs to be fixed by SOGo.

**Comment by rawvoid** [comment]
But there is no problem feedback on the [github homepage](https://github.com/Alinto/sogo) of the SOGo project.

**Comment by tarvfx** [comment]
Can mailcow add a configuration option, administrators can use regular expressions to configure which characters are legal.

**Comment by tarvfx** [comment]
Can mailcow add a configuration option, which administrators can use regular expressions to configure which characters are legal?

**Comment by tarvfx** [comment]
Can mailcow add a configuration option so that administrators can use regular expressions to configure which characters are legal?

**Comment by rawvoid** [comment]
Can mailcow add a configuration option so that administrators can use regex to limit which characters are legal?

**Comment by MAGICCC** [comment]
> But there is no problem feedback on the [github homepage](https://github.com/Alinto/sogo) of the SOGo project.

They have their bugtracker here https://bugs.sogo.nu

**Comment by FreddleSpl0it** [comment]
> The cause of the problem may be that the '/' character in the URL redirected by SOGo after successful login is not URL-encoded. Maybe this problem needs to be fixed by SOGo.

There are more problems with mailboxes containing `/`. Try to edit the mailbox. mailcow will display `No action applicable`.
I also think that such mailboxes cannot even log in to the mailcow UI.
On mailbox creation, it checks for
https://github.com/mailcow/mailcow-dockerized/blob/8f286669162805f74e128ae20fabb15bb49c538d/data/web/inc/functions.mailbox.inc.php#L2803

And on login
https://github.com/mailcow/mailcow-dockerized/blob/8f286669162805f74e128ae20fabb15bb49c538d/data/web/inc/functions.inc.php#L818

**Comment by milkmaker** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Comment by patschi** [comment]
Solved as per #5619
